File: src/test/java/net/slipp/domain/qna/AnswerBuilder.java
package net.slipp.domain.qna;

import net.slipp.domain.user.SocialUser;

public class AnswerBuilder {
	private Long answerId = 0L;
	private int totalLiked = 0;
	private SocialUser writer;

	private AnswerBuilder() {}

	private AnswerBuilder(Long answerId) {
		this.answerId = answerId;
	}
	
	public static AnswerBuilder anAnswer() {
		return new AnswerBuilder();
	}

	public static AnswerBuilder anAnswer(Long answerId) {
		return new AnswerBuilder(answerId);
	}
	
	public AnswerBuilder withTotalLiked(int totalLiked) {
		this.totalLiked = totalLiked;
		return this;
	}
	
	public AnswerBuilder with(SocialUser writer) {
		this.writer = writer;
		return this;
	}
	
	public Answer build() {
		Answer answer = new Answer() {
			@Override
			public Integer getSumLike() {
				return totalLiked;
			}
		};
		answer.setAnswerId(answerId);
		answer.writedBy(writer);
		return answer;
	}
}

File: src/test/scala/net/slipp/domain/qna/AnswerBuilder.scala
package net.slipp.domain.qna

import java.lang.Long
import net.slipp.domain.user.SocialUser

object AnswerBuilder {
  def anAnswer: AnswerBuilder = {
    return new AnswerBuilder(null)
  }

  def anAnswer(answerId: Long): AnswerBuilder = {
    return new AnswerBuilder(answerId)
  }
}

class AnswerBuilder(answerId: Long){
  private var totalLiked: Int = 0
  private var writer: SocialUser = null

  def withTotalLiked(totalLiked: Int): AnswerBuilder = {
    this.totalLiked = totalLiked
    return this
  }

  def `with`(writer: SocialUser): AnswerBuilder = {
    this.writer = writer
    return this
  }

  def build: Answer = {
    val answer: Answer = new Answer() {
      override def getSumLike: Integer = {
        return totalLiked
      }
    }
    answer.setAnswerId(answerId)
    answer.writedBy(writer)
    return answer
  }
}
